katportalclient/client.py

```diff
@@ -1254,73 +1196,37 @@ def sensor_history(self, sensor_name, start_time_sec, end_time_sec,
             - If there was an error submitting the request.
             - If the request timed out
         """
-        # create new namespace and state variables per query, to allow multiple
-        # request simultaneously
-        state = {
-            'sensor': sensor_name,
-            'done_event': tornado.locks.Event(),
-            'num_samples_pending': 0,
-            'include_value_ts': include_value_ts,
-            'samples': []
-        }
-        namespace = str(uuid.uuid4())
-        self._sensor_history_states[namespace] = state
-        # ensure connected, and subscribed before sending request
-        yield self.connect()
-        yield self.subscribe(namespace, ['*'])
-
         params = {
             'sensor': sensor_name,
-            'time_type': SAMPLE_HISTORY_REQUEST_TIME_TYPE,
-            'start': start_time_sec * SAMPLE_HISTORY_REQUEST_MULTIPLIER_TO_SEC,
-            'end': end_time_sec * SAMPLE_HISTORY_REQUEST_MULTIPLIER_TO_SEC,
-            'namespace': namespace,
-            'request_in_chunks': 1,
-            'chunk_size': SAMPLE_HISTORY_CHUNK_SIZE,
-            'limit': MAX_SAMPLES_PER_HISTORY_QUERY
+            'start_time': start_time_sec,
+            'end_time': end_time_sec,
+            'limit': MAX_SAMPLES_PER_HISTORY_QUERY,
+            'timeout': timeout_sec
         }
+        if additional_fields: 
+            params['additional_fields'] = additional_fields
```

Comment:
Suggest this block is just something like:
```
if include_value_ts:
    additional_fields = 'status,value_time'
else:
    additional_fields = 'status'
```
(Not sure how a list is specified for additional fields)